lib/core/include/client_connection.hpp

```diff
@@ -52,6 +52,25 @@ namespace irods::experimental
                           const std::string_view _username,
                           const std::string_view _zone);
 
+        /// Connects to the iRODS server identified by the passed arguments
+        /// and authenticates the proxy user using the information and credentials
+        /// found in the user's irods_environment.json file
+        ///
+        /// \throws irods::exception If an error occured.
+        ///
+        /// \param[in] _host     The host name of the iRODS server.
+        /// \param[in] _port     The port to connect to.
+        /// \param[in] _proxy_user The name of the user to proxy
+        /// \param[in] _proxy_zone The name of the zone to proxy
+        /// \param[in] _username The user to connect as.
+        /// \param[in] _zone     The zone managed by the iRODS server.
```

Comment:
add `\since 4.3.0` line just like the other functions

please vertically align the parameter descriptions like the other functions